RevenueCatUI/Templates/V2/Components/Tabs/TabsComponentView.swift

```diff
@@ -0,0 +1,703 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TabsComponentView.swift
+//
+//  Created by Josh Holtz on 1/9/25.
+// swiftlint:disable file_length
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+class TabControlContext: ObservableObject {
+
+    @Published
+    var selectedIndex: Int = 0
+
+    let controlStackViewModel: StackComponentViewModel
+    let tabControlStackViewModels: [StackComponentViewModel]
+
+    init(controlStackViewModel: StackComponentViewModel,
+         tabControlStackViewModels: [StackComponentViewModel]) {
+        self.controlStackViewModel = controlStackViewModel
+        self.tabControlStackViewModels = tabControlStackViewModels
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TabsComponentView: View {
+
+    @EnvironmentObject
+    private var packageContext: PackageContext
+
+    private let viewModel: TabsComponentViewModel
+    private let onDismiss: () -> Void
+
+    init(viewModel: TabsComponentViewModel, onDismiss: @escaping () -> Void) {
+        self.viewModel = viewModel
+        self.onDismiss = onDismiss
+    }
+
+    var body: some View {
+        LoadedTabsComponentView(
+            viewModel: self.viewModel,
+            parentPackageContext: self.packageContext,
+            onDismiss: self.onDismiss
+        )
```

Comment:
Environment variables cannot be referenced in initializers so... I created `LoadedTabsComponentView` so that it would be passed `self.packageContext` since we need a the list of packages for creating a context for each tab/tier